lib/pipeline.js

```diff
@@ -1658,6 +1659,75 @@ class PipelineModel extends BaseModel {
         return factory.list(listConfig);
     }
 
+    /**
+     * Fetch builds belonging to a pipeline or events
+     * @param  {Object}   [config]
+     * @param  {Number}   [config.sort]                         Sort rangekey by ascending or descending
+     * @param  {Number}   [config.sortBy]                       Sortby field
+     * @param  {Object}   [config.paginate]                     Pagination parameters
+     * @param  {Number}   [config.paginate.count]               Number of items per page
+     * @param  {Number}   [config.paginate.page]                Specific page of the set to return
+     * @param  {Number}   [config.groupEventId]                 Group event ID
+     * @param  {Boolean}  [config.latest]                       If we want to return latest in groupEventId, default false
+     * @return {Promise}  Resolves to an array of builds
+     */
+    async getBuilds(config = {}) {
+        const { sort } = config;
+        const latest = hoek.reach(config, 'params.latest');
+        const groupEventId = hoek.reach(config, 'params.groupEventId');
+
+        // Fetch all builds for each event with same groupEventId
+        if (groupEventId && !latest) {
+            // Lazy load factory dependency to prevent circular dependency issues
+            // https://nodejs.org/api/modules.html#modules_cycles
+            /* eslint-disable global-require */
+            const EventFactory = require('./eventFactory');
+            /* eslint-enable global-require */
+            const eventFactory = EventFactory.getInstance();
+            const events = await eventFactory.list({
+                params: { groupEventId }
+            });
+            const processed = [];
+
+            events.forEach(e => processed.push(e.getBuilds()));
+
+            return Promise.all(processed).then(builds => {
+                // flatten array of arrays
+                return builds.flat(1);
+            });
+        }
+        // Lazy load factory dependency to prevent circular dependency issues
+        // https://nodejs.org/api/modules.html#modules_cycles
+        /* eslint-disable global-require */
+        const BuildFactory = require('./buildFactory');
+        /* eslint-enable global-require */
+        const buildFactory = BuildFactory.getInstance();
+
+        // Fetch only latest builds with same groupEventId
+        if (latest && groupEventId) {
+            return buildFactory.getLatestBuilds({
+                groupEventId: config.params.groupEventId
+            });
+        }
+
+        // Latest should not be passed to list config
+        if (latest !== undefined) {
+            delete config.params.latest;
+        }
+
+        // Fetch builds for this pipeline with default count and page
+        const defaultConfig = {
+            params: { pipelineId: this.id },
+            paginate: {
+                count: DEFAULT_COUNT
+            },
```

Comment:
if page is not implicitly defined, it is defaulted to `1`? if yes, then maybe comment should be worded as such:
`// Fetch builds for this pipeline with default count and page (implicitly set to 1)`

Reply:
added